### DISCUSSION TITLE
Add an extension to a Keycloak docker testcontainer

### DISCUSSION THREAD

**Discussion by swc-kdzekov** [dasniko/testcontainers-keycloak#50] [General]
Hello,
I would appreciate a tip or advice here.
I'm writing an integration test that is using keycloak (docker) container. I would like to add an extension (jar file) to it. In the real docker environment the .yaml file has volumes defined:

`volumes:     - C:\keycloak\plugins\default-settings-event-listener-0.0.1-SNAPSHOT.jar:/opt/jboss/keycloak/standalone/deployments/default-settings-event-listener-0.0.1-SNAPSHOT.jar `

In the test the container is defined as:
```
 KeycloakContainer KC_CONTAINER = new KeycloakContainer()
                .withAdminUsername("username")
                .withAdminPassword("password");


 KC_CONTAINER.start();
```
I would like to know how can I add the jar file (keeping it in resources folder in my project) into the container?

I know there are some methods like: setVolumesFroms(), withVolumesFroms(), but I haven't found proper documentation on how to use them.

Thank you!

**Comment by dasniko** [comment]
If you want to test your custom extension, you can add the compiled output (`target/classes` folder, not `.jar`!) like documented here: https://github.com/dasniko/testcontainers-keycloak#testing-custom-extensions

**Comment by swc-kdzekov** [comment]
Thank you!
